### DISCUSSION TITLE
Histogram Output Oddness

### DISCUSSION THREAD

**Discussion by blue-j** [ImageMagick/ImageMagick#2799] [General]
Hi everyone! I hope you are well during this crazy time. I'm having an oddity with histogram output:

`convert in.jpg  -alpha off -depth 8 -flatten -separate -append -strip -define histogram:unique-colors=false histogram:histo.gif`

is outputting a fine global histogram, except it has a huge spike in the bin of pure white all the way to the top. It seems this occurs no matter what image I use as input! (I would provide a sample but believe it is irrelevant and will happen for you, too! Please tell me if I'm wrong of course.) I just tried a simple JPG with no odd features, no alphas, no paths, nothing odd.

My build, installed with Homebrew on macOS 10.15.7:

ImageMagick 7.0.10-34 Q16 x86_64 2020-10-09
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib

**Comment by fmw42** [comment]
I see that also on 7.0.10-35 Q16 Mac OSX Sierra. Tested your command with rose: image.

**Comment by snibgo** [comment]
The problem also occurs in v7.0.10-27 on Windows 8.1.

In v7.0.8-64, the problem does not occur.

**Comment by blue-j** [comment]
Thanks for checking!

(I suspect it's some issue with the 8-bit range being shifted - instead of 0-255, maybe 1-256 or something along those lines).

**Reply by blue-j** [reply]
(I have made it into a bug report)

**Reply by blue-j** [reply]
(It was the order of the arguments that created the problem.  IM was and is working correctly.)
